packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/task/ViewGanttDescriptionConverter.java

```diff
@@ -0,0 +1,147 @@
+import java.util.Optional;
+import java.util.function.Function;
+
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.sirius.components.compatibility.emf.DomainClassPredicate;
+import org.eclipse.sirius.components.core.api.IObjectService;
+import org.eclipse.sirius.components.gantt.TaskDetail;
+import org.eclipse.sirius.components.gantt.description.GanttDescription;
+import org.eclipse.sirius.components.gantt.description.TaskDescription;
+import org.eclipse.sirius.components.interpreter.AQLInterpreter;
+import org.eclipse.sirius.components.representations.GetOrCreateRandomIdProvider;
+import org.eclipse.sirius.components.representations.IRepresentationDescription;
+import org.eclipse.sirius.components.representations.VariableManager;
+import org.eclipse.sirius.components.view.RepresentationDescription;
+import org.eclipse.sirius.components.view.emf.IRepresentationDescriptionConverter;
+import org.springframework.stereotype.Service;
+
+/**
+ * Converts a View-based gantt description into an equivalent {@link GanttDescription}.
+ *
+ * @author lfasani
+ */
+@Service
+public class ViewGanttDescriptionConverter implements IRepresentationDescriptionConverter {
+
+    private static final String DEFAULT_GANTT_LABEL = "Gantt";
+
+    private static final String DEFAULT_GANTT_DESCRIPTION_LABEL = "Gantt Description";
+
+    private final IObjectService objectService;
+
+    private final Function<VariableManager, String> semanticTargetIdProvider;
+
+    private final Function<VariableManager, String> semanticTargetKindProvider;
+
+    private final Function<VariableManager, String> semanticTargetLabelProvider;
+
+    private GanttIdProvider ganttIdProvider;
+
+    public ViewGanttDescriptionConverter(IObjectService objectService, GanttIdProvider ganttIdProvider) {
+        this.objectService = Objects.requireNonNull(objectService);
+        this.ganttIdProvider = ganttIdProvider;
+        this.semanticTargetIdProvider = variableManager -> this.self(variableManager).map(this.objectService::getId).orElse(null);
+        this.semanticTargetKindProvider = variableManager -> this.self(variableManager).map(this.objectService::getKind).orElse(null);
+        this.semanticTargetLabelProvider = variableManager -> this.self(variableManager).map(this.objectService::getLabel).orElse(null);
+
+    }
+
+    @Override
+    public boolean canConvert(RepresentationDescription representationDescription) {
+        return representationDescription instanceof org.eclipse.sirius.components.view.gantt.GanttDescription;
+    }
+
+    @Override
+    public IRepresentationDescription convert(RepresentationDescription viewRepresentationDescription, List<RepresentationDescription> allRepresentationDescriptions, AQLInterpreter interpreter) {
+        final org.eclipse.sirius.components.view.gantt.GanttDescription viewGanttDescription = (org.eclipse.sirius.components.view.gantt.GanttDescription) viewRepresentationDescription;
```

Comment:
Don't mark local variables as final, there is really no need for that

Reply:
done